tests/system_test.py

```diff
@@ -114,6 +115,39 @@ def _check_requirements():
 MISSING_REQUIREMENTS = _check_requirements()
 
 
+def flakey(issue: str, repeats: int = 3):
+    """Decorator that marks test as flakey (flaky).
+
+    If applied, executes the test up to three times, marking it as failed only if it fails each time.
+    Note, that use of this decorator is generally discouraged---tests should pass reliably when their assertions are upheld.
+
+    Example usage
+
+        @flakey(issue='#123')
+        def test_feature_in_fragile_manner():
+            self.assertTrue(...)
+    """
+    del issue  # unused
```

Comment:
One thing to consider is to force the use of the issue parameter so we're certain there is a corresponding GH issue for the test.  Keeps us honest. And probably worth including the issue in the logging/assertion error text.